src/helpers/keyManager/ReactNativeKeychainFacade.ts

```diff
@@ -0,0 +1,235 @@
+  public async setKey(id: string, key: EncryptedKey): Promise<void> {
+    const storageKey = sanitizeKey(id);
+    try {
+      await Keychain.setGenericPassword(storageKey, JSON.stringify(key), {
+        service: `${this.service}_${storageKey}`,
+      });
+    } catch (error) {
+      logger.error(`Error setting key ${id}:`, error);
+      throw new Error(`Failed to set key ${id}`);
+    }
+  }
+
+  /**
+   * Remove a key from keychain
+   */
+  public async removeKey(id: string): Promise<void> {
+    const key = sanitizeKey(id);
+    try {
+      await Keychain.resetGenericPassword({
+        service: `${this.service}_${key}`,
+      });
+    } catch (error) {
+      logger.error(`Error removing key ${id}:`, error);
+      // Don't throw here, as the key might not exist
+    }
+  }
+
+  /**
+   * Get all keys from keychain
+   * Uses the special index key to track all stored keys
+   */
+  public async getAllKeys(): Promise<EncryptedKey[]> {
+    const keys: EncryptedKey[] = [];
+    try {
+      // Get the index key with service only
+      const result = await Keychain.getGenericPassword({
+        service: `${this.service}_index`,
+      });
+
+      if (result === false) {
+        return [];
+      }
+
+      const keyIds: string[] = JSON.parse(result.password);
+
+      for (const id of keyIds) {
+        const key = await this.getKey(id);
+        if (key !== null) {
+          keys.push(key);
+        }
+      }
+    } catch (error) {
+      logger.error("Error getting all keys:", error);
+      // No index found, return empty array
+    }
+    return keys;
+  }
+
+  /**
+   * Add a key ID to the index
+   */
+  public async addToKeyIndex(id: string): Promise<void> {
+    try {
+      // Using a specific service name for the index
+      const result = await Keychain.getGenericPassword({
+        service: `${this.service}_index`,
+      });
+
+      let keyIds: string[] = [];
+
+      if (result !== false) {
+        keyIds = JSON.parse(result.password);
+      }
+
+      if (!keyIds.includes(id)) {
+        keyIds.push(id);
+        await Keychain.setGenericPassword("index", JSON.stringify(keyIds), {
+          service: `${this.service}_index`,
+        });
+      }
+    } catch (error) {
+      // If index doesn't exist yet, create it
+      logger.error("Error adding to key index, creating new index:", error);
+      await Keychain.setGenericPassword("index", JSON.stringify([id]), {
+        service: `${this.service}_index`,
+      });
+    }
+  }
+
+  /**
+   * Remove a key ID from the index
+   */
+  public async removeFromKeyIndex(id: string): Promise<void> {
+    try {
+      const result = await Keychain.getGenericPassword({
+        service: `${this.service}_index`,
+      });
+
+      if (result !== false) {
+        const keyIds: string[] = JSON.parse(result.password);
+        const newKeyIds = keyIds.filter((keyId) => keyId !== id);
+
+        await Keychain.setGenericPassword("index", JSON.stringify(newKeyIds), {
```

Comment:
looks like we are using this "index" username in a few places, should we move it to a const?